streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -250,10 +250,11 @@ StampedRecord nextRecord(final RecordInfo info, final long wallClockTime) {
 
         if (queue != null) {
             // get the first record from this queue.
+            final int oldSize = queue.size();
             record = queue.poll(wallClockTime);
 
             if (record != null) {
-                --totalBuffered;
+                totalBuffered -= (oldSize - queue.size());
```

Comment:
This is a follow up fix: when we skip more than one corrupted record, we don't update the buffer size correctly.